m4/gx_fortran_options.m4

```diff
@@ -255,23 +255,28 @@ AC_LANG_POP([Fortran])
 # action-if-found, otherwise execute action-if-not-found.  If only is specified
 # then check if the Fortran module has the given symbol.
 AC_DEFUN([GX_FC_CHECK_MOD],[
-AC_LANG_PUSH([Fortran])
+_AC_FORTRAN_ASSERT()dnl
 m4_ifval([$2],[gx_fc_check_mod_only=",only:$2"],[gx_fc_check_mod_only=""])
 AS_LITERAL_WORD_IF([$1],
   [AS_VAR_PUSHDEF([gx_mod], [gx_cv_fc_check_mod_$1])],
   [AS_VAR_PUSHDEF([gx_mod], AS_TR_SH([gx_cv_check_mod_$1]))])
+dnl Autoconf does not pass CPPFLAGS to the Fortran tests.  As the user may
+dnl define the include options in CPPFLAGS instead of FFLAGS or FCFLAGS, we
+dnl force CPPFLAGS te be part of FFLAGS/FCFLAGS
+gx_save_[]_AC_LANG_PREFIX[]FLAGS=${[]_AC_LANG_PREFIX[]FLAGS}
+_AC_LANG_PREFIX[]FLAGS="${[]_AC_LANG_PREFIX[]FLAGS} $CPPFLAGS"
```

Comment:
Disagree with this plan, it just adds confusion.

Instead, properly educate your users on FCFLAGS vs. CPPFLAGS.

Reply:
I agree that in this case (when checking for Fortran `.mod` files) adding `CPPFLAGS` to `FCFLAGS` or `FFLAGS` is not a solution for all Fortran packages.  However, in our case, all Fortran files are run through the preprocessor.  Automake will pass `CPPFLAGS` when compiling preprocessed Fortran.  Thus in this case, we are simply forcing the configure tests to mimic the behavior of automake.

Reply:
Yes, but in many cases users will have already added the `-I...` compile directive for libraries (like netCDF) in the `CPPFLAGS` for C.  On many of our systems, the C and Fortran include files are in the same directories.  Users can use one or both. My goal was to have the configure tests mimic what automake does.